Add debugging steps for DiscoverEC2 User Task issues (#47513)
gravitational/teleport: api/types/usertasks/descriptions/ec2-ssm-invocation-failure.md

Teleport failed to access the SSM Agent to auto enroll the instance.

> This is not a failure of the script that ran in the instance, but rather a failure when trying to communicate with the AWS Systems Manager service.

Reviewer: ```suggestion
Some instances failed to communicate with the AWS Systems Manager service to execute the install script.
```

Usually this happens when:

**Missing policies**

The IAM Role used by the integration might be missing some required permissions.
Ensure the following actions are allowed in the IAM Role used by the integration:
- `ec2:DescribeInstances`
- `ssm:DescribeInstanceInformation`
- `ssm:GetCommandInvocation`
- `ssm:ListCommandInvocations`
- `ssm:SendCommand`

**SSM Document is invalid**

> Teleport uses an SSM Document to run an installation script.

Reviewer: How/where can a user find this document?

Author: I'll the SSM Document name to the list of fields.
UI should then build the SSM Document URL (we have the region and the SSM Document name, so that should be simple).

Reviewer: @marcoandredinis I think these messages should be templated using text/template. Then whatever returns these messages to the web UI can parameterize them with appropriate data without having to rely on the web UI to assemble a bunch of stuff.

Author: We can template everything or we show have specific fields for metadata and then only the description is templated.

Taking the following screen as example
![image](https://github.com/user-attachments/assets/79bebbdd-1251-49c4-9472-44a000f82a31)

I was thinking that only the `Details` section was to be converted from Markdown into HTML (aka sent over the API).

After reading your comment I wonder if you want it all to be generated in the backend.
Which parts should be rendered in the backend?
Title
Metadata (region, account id)
Details
Impacted instances

We can do it all, if that's what we want.
However, we have to be careful because there are some elements that might look out of place (using a different CSS style) because the library we'll be using will not use the same set of styles that we have.
While this might be easy to fix for things that are mostly text (bolds, lists and such), it might not be easy for more complex elements like tables.

(@kimlisa please correct me if I'm wrong because I don't know the library we'll be using)

Reviewer 2: React-markdown sounds OK, but we don't actually use it right now since Assist is deprecated. 

Is there a reason to send markdown to the front end, vs sending an error code indicating which display to show, and the UI has a collection of views?

Reviewer: @michellescripts The reason I'd prefer for the frontend to just be a simpler renderer in this case is so we can easily expand our "library" of errors (with descriptions and other metadata) without having to update UI every time we want to add a new type of user task.

Reviewer 2: I'm fine with that! I do agree that templating in links for the user will be a better UX but we can tackle that whenever.

Author: We have 4 sections: title, metadata, description and table with affected resources.

My suggestion:
- Title is just a string.
- Metadata should come from the backend and separate fields and rendered in the frontend.
- Description comes from the backend as markdown and is rendered in the frontend using react-markdown.
- Table comes as a list from the backend, and is rendered in the frontend.
